File: Data Structures & Algorithms/missing-number/submission-0.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        # realSum = (len(nums) * (len(nums) + 1))//2
        # return realSum - sum(nums)

        res = nums[0]
        for n in range(1, len(nums)):
            res ^= nums[n]

        for i in range(len(nums) + 1):
            res ^= i

        return res

File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int bucket[3] = {0, 0, 0};
        for (int elem : nums) {
            bucket[elem]++;
        }

        int i = 0;
        int j = 0;
        while (i < nums.size()) {
            while (bucket[j]) {
                nums[i] = j;
                bucket[j]--;
                i++;
            }
            j++;
        }
    }
};